src/dnode/src/dnodeMain.c

```diff
@@ -135,6 +152,24 @@ static void signal_handler(int32_t signum, siginfo_t *sigInfo, void *context) {
   exit(EXIT_SUCCESS);
 }
 
+static void dnodeCleanupSteps(int32_t stepId) {
```

Comment:
rename "dnodeCleanupSteps" to "dnodeCleanupComponents". In addition, remove the input parameter "stepId", it shall be similar to other init/cleanup functions

Reply:
The function introduces a *stepId* because while it cleanup during failure, you don't want it to run cleanup() function where its corresponding init() never runs. For example if the readInit fails, you only want the unwinding starts from readCleanup and backwards.

Reply:
Yeah, a good way to handle this scenario